Examples/SmokeTest/SmokeTest/ViewInstrumentationView.swift

```diff
@@ -0,0 +1,98 @@
+import Foundation
+import OpenTelemetryApi
+import SwiftUI
+import UIKit
+
+@testable import Honeycomb
+
+private struct NestedExpensiveView: View {
+    let delay: Double
+
+    var body: some View {
+        HStack {
+            HoneycombInstrumentedView(name: "nested expensive text") {
+                Text(String(timeConsumingCalculation(delay)))
+            }
+        }
+    }
+}
+
+private struct ExpensiveView: View {
+    @State private var delay = 2.0
+    @State private var sliderDelay = 2.0
+    @State private var isEditing = false
+
+    var body: some View {
+        HoneycombInstrumentedView(name: "main view") {
+            VStack {
+                Spacer()
+
+                Slider(
+                    value: $sliderDelay,
+                    in: 0...4,
+                    step: 0.5
+                ) {
+                    Text("Delay")
+                } minimumValueLabel: {
+                    Text("0")
+                } maximumValueLabel: {
+                    Text("4")
+                } onEditingChanged: { editing in
+                    isEditing = editing
+                    if !editing {
+                        delay = sliderDelay
```

Comment:
I really thought that since the value of the slider was a `$` binding that you would just be able to have `Slider(value: $delay,` and not have any `onEditingChanged:` handler. If this is necessary, could you add a comment explaining why we need all three state variables?

Reply:
yeah the idea was that we'd be able to adjust the slider but the rest of the view wouldn't re-render until you stopped editing it. 

BUT clearly I wasn't paying enough attention and that wasn't actually working. I'll restructure it and add a comment.

Reply:
works now:

https://github.com/user-attachments/assets/eb5e5d50-e45d-4698-a9d9-f158c90183ac

(yes i know the simulator has a screen recorder but i wanted to capture my mouse cursor lol)